File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    cntx=[]
    cnty=[]
    comm=[]
    for i in range(10):

        if X.count(str(i))!=0 and Y.count(str(i))!=0 :
            for j in range(min(X.count(str(i)),Y.count(str(i)))):
                comm.append(str(i))
                
    if not comm: return "-1"
    comm.sort(reverse=True)
    answer=''.join(comm)
    if answer[-1]=='0'and answer[0]=='0':return "0"
    
    return answer

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
class Solution {
    public String solution(String X, String Y) {
        String answer = "";
        
        int[] cntx=new int[10];
        int[] cnty=new int[10];
        String[] x=X.split("");
        String[] y=Y.split("");
        
        for(String n:x) cntx[Integer.parseInt(n)]++;
        for(String n:y) cnty[Integer.parseInt(n)]++;
        
        for(int i=9;i>=0;i--){
            int l=cnty[i];
            int s=cntx[i];
            if(cntx[i]>cnty[i]){
                l=cntx[i];
                s=cnty[i];
            }
            answer+=(i+"").repeat(s);
        }
        
        if(answer.equals("")) answer="-1";
        else if(answer.charAt(0)=='0') answer="0";
        
        
        return answer;
    }
}